features/object_mapper/using/README.md

```diff
@@ -0,0 +1,267 @@
+
+Here is an example with the previously defined `User` class :
+
+```java
+@Accessor
+public interface UserAccessor {
+    @Query("SELECT * FROM user")
+    Result<User> getAll();
+}
+```
+
+The return type is important here, the results of the query will be automatically either mapped in a single object or in a collection of objects.
+
+The `MappingManager` can then process this class and generate an implementation :
+
+```java
+UserAccessor userAccessor = manager.createAccessor(UserAccessor.class);
+User user = userAccessor.getOne(uuid);
+```
+
+Like mappers, accessors are cached at the manager level and thus, are thread-safe/sharable.
+
+--------------
+
+Named or unnamed bind markers can be used in a `@Query`, when using a bind marker in a `@Query`, the corresponding accessor's method can accept parameters. Parameters are specified with `@Param` annotation before each parameter of the method : 
+
+```java
+@Accessor
+public interface UserAccessor {
+    @Query("SELECT * FROM user WHERE user_id=?")
+    User getOne(UUID userId);
+}
+
+@Accessor
+public interface FooAccessor {
+    @Query("insert into foo (userId, name) values (:u, :n)")
+    ResultSet insert(@Param("u") UUID userId, @Param("n") String name);
+}
+
+@Accessor
+public interface FooBindMarkerAccessor {
+    @Query("insert into foo (k, v) values (?, ?)")
+    ResultSet insert(@Param("u") UUID userId, @Param("n") String name);
+}
+```
+
+--------------
+
+A method parameter can be preceded by `@Enumerated` instead of `@Param`, this to state that the parameter is a Java enumeration. As for enumeration fields, the `EnumType` has to be specified.
+
+The following example acts on two tables : `user(key int primary key, gender int)` and `user_str(key int primary key, gender text)`.
+
+```java
+enum Enum {
+    MALE, FEMALE
+}
+
+@Table(name = "user")
+public static class User {
+    @PartitionKey
+    private int key;
+
+    @Enumerated(EnumType.ORDINAL)
+    private Enum gender;
+    
+    // ... constructors / getters / setters
+}
+
+@Table(name = "user_str")
+public static class UserStr {
+    @PartitionKey
+    private int key;
+
+    private Enum gender;
+    
+    // ... constructors / getters / setters
+}
+
+@Accessor
+public interface UserAccessor {
+    @Query("insert into user (key, gender) values (?,?)")
+    ResultSet addUser(int key, @Enumerated(EnumType.ORDINAL) Enum value);
+    
+    @Query("insert into user_str (key, gender) values (?,?)")
+    ResultSet addUserStr(int key, @Enumerated(EnumType.STRING) Enum value);
+}
+```
+
+--------------
+
+To specify whether the queries in a `Accessor` need to be executed asynchronously, the return type has to be set as a `ListenableFuture` :
```

Comment:
Would be good to document all possible return types, see http://docs.datastax.com/en/developer/java-driver/2.1/common/drivers/reference/accessorAnnotatedInterfaces.html